deps/CMakeLists.txt

```diff
@@ -15,16 +15,16 @@ endfunction()
 
 execute_process(
     COMMAND git submodule update --init
-    WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/deps
 )
 
-mapbox_base_add_library(geojson-vt-cpp ${CMAKE_CURRENT_LIST_DIR}/geojson-vt-cpp/include)
-mapbox_base_add_library(geojson.hpp ${CMAKE_CURRENT_LIST_DIR}/geojson.hpp/include)
-mapbox_base_add_library(geometry.hpp ${CMAKE_CURRENT_LIST_DIR}/geometry.hpp/include)
-mapbox_base_add_library(jni.hpp ${CMAKE_CURRENT_LIST_DIR}/jni.hpp/include)
-mapbox_base_add_library(optional ${CMAKE_CURRENT_LIST_DIR}/optional)
-mapbox_base_add_library(pixelmatch-cpp ${CMAKE_CURRENT_LIST_DIR}/pixelmatch-cpp/include)
-mapbox_base_add_library(shelf-pack-cpp ${CMAKE_CURRENT_LIST_DIR}/shelf-pack-cpp/include)
-mapbox_base_add_library(supercluster.hpp ${CMAKE_CURRENT_LIST_DIR}/supercluster.hpp/include)
-mapbox_base_add_library(variant ${CMAKE_CURRENT_LIST_DIR}/variant/include)
-mapbox_base_add_library(cheap-ruler-cpp ${CMAKE_CURRENT_LIST_DIR}/cheap-ruler-cpp/include)
+mapbox_base_add_library(geojson-vt-cpp ${PROJECT_SOURCE_DIR}/deps/geojson-vt-cpp/include)
```

Comment:
This is similar like to use absolute path in a script, instead of relative. Every time when something change in a structure of repo you will nee to fix all these lines

Reply:
@deniskoronchik this is not going to happen so frequently, right? Also, we need to be careful with list, source directory usage in CMake: some projects include this, some others add it as subdirectory, so the only variable that always remains correct is `PROJECT_SOURCE_DIR` given the fact `mapbox-base` has a project described in its root source directory.

Reply:
But why you decided that our team doesn't need that?
How we can check all libraries on our target device in TME project?

Reply:
`CMAKE_CURRENT_LIST_DIR` - is a path to a directory, where exactly this file placed
https://cmake.org/cmake/help/v3.0/variable/CMAKE_CURRENT_LIST_DIR.html

```
As CMake processes the listfiles in your project this variable will always be set to the directory where the listfile which is currently being processed (CMAKE_CURRENT_LIST_FILE) is located
```

Reply:
See https://stackoverflow.com/a/15667036 on why this is not always true.

Reply:
In which cases we need to add it as `include`, but not `add_subdirectory`?

Reply:
> In which cases we need to add it as include, but not add_subdirectory?

It depends, but based on prior experience, people often misuse these, and thus we need to come up with a robust solution that supports all cases.